File: frontend/src/lib/deploy.ts
import { api } from "@/lib/api";

export type ProductionStatus = {
  branch: string;
  commit: string;
  dirty: boolean;
  coolify_health: Record<string, unknown>;
  coolify_api: Record<string, unknown>;
  ollama_ok: boolean;
  ollama_health: Record<string, unknown>;
};

export type CoolifyDeployResult = {
  configured: boolean;
  triggered: boolean;
  status_code: number | null;
  body: string | null;
};

export async function fetchProductionStatus() {
  const response = await api.get<ProductionStatus>("/api/v1/deploy/coolify/status");
  return response.data;
}

export async function triggerCoolifyDeploy() {
  const response = await api.post<CoolifyDeployResult>("/api/v1/deploy/coolify");
  return response.data;
}

File: backend/app/schemas/deploy.py
from typing import Any

from pydantic import BaseModel


class ProductionStatus(BaseModel):
    branch: str
    commit: str
    dirty: bool
    coolify_health: dict[str, Any]
    coolify_api: dict[str, Any]
    ollama_ok: bool
    ollama_health: dict[str, Any]


class CoolifyDeployResult(BaseModel):
    configured: bool
    triggered: bool
    status_code: int | None = None
    body: str | None = None
